WcaOnRails/lib/results_validators/persons_validator.rb

```diff
@@ -65,8 +65,8 @@ def validate(competition_ids: [], model: Result, results: nil)
         end
 
         without_wca_id, with_wca_id = persons_by_id.values.partition { |p| p.wca_id.empty? }
-        if without_wca_id.any?
-          existing_person_in_db_by_name = Person.where(name: without_wca_id.map(&:name)).group_by(&:name)
+        existing_person_in_db_by_name = Person.where("LOWER(name) IN (?)", without_wca_id.map(&:name).map(&:downcase)).group_by(&:name)
+        if existing_person_in_db_by_name.any?
```

Comment:
Not sure whether an `if any?` check is necessary here. The next line does `.each` which simply won't loop if the result array is empty.